integrations.go

```diff
@@ -67,39 +63,15 @@ func getModules() (map[string]string, error) {
 	return nil, fmt.Errorf("module integration failed")
 }
 
-func getModulesFromMod() (map[string]string, error) {
-	modules := make(map[string]string)
-
-	file, err := os.Open("go.mod")
-	if err != nil {
-		return nil, fmt.Errorf("unable to open mod file")
-	}
-
-	defer file.Close()
-
-	areModulesPresent := false
-
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		splits := strings.Split(scanner.Text(), " ")
-
-		if splits[0] == "require" {
-			areModulesPresent = true
-
-			// Mod file has only 1 dependency
-			if len(splits) > 2 {
-				modules[strings.TrimSpace(splits[1])] = splits[2]
-				return modules, nil
-			}
-		} else if areModulesPresent && splits[0] != ")" && splits[0] != "" {
-			modules[strings.TrimSpace(splits[0])] = splits[1]
-		}
+func getModulesFromBuildInfo() (map[string]string, error) {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return nil, fmt.Errorf("cannot dependencies from debug.BuildInfo")
 	}
-
-	if scannerErr := scanner.Err(); scannerErr != nil {
-		return nil, scannerErr
+	modules := make(map[string]string)
+	for _, dep := range info.Deps {
+		modules[dep.Path] = dep.Version
```

Comment:
Needs to include `info.Main` too 😉 

---

As you well noted in a comment, we also need to handle replacements. The Sentry server only takes a `map[string]string` for package/module name => version, no concept of replacements. While we could look into adding replacements server-side and in the web interface, I think we can well start with just reporting replacements within the current server constraints. 

There are replacements pointing to a module+version, and replacements pointing to absolute local path and relative local path, see https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive.

We could use a syntax similar to how `go list -m all` prints replacements:

```
module/name v0.0.0 => replacement
```

```go
map[string]string{"module/name": "v0.0.0 => replacement"}
```

What do you think?

Reply:
I think it sound good to me for doing this.

Reply:
Fixed.